### PR TITLE
Update to Mbed Cloud Client 2.0.1

### PR DESCRIPTION
**Pull request by janjongboom**
Follow up from #27. This updates Simple Pelion DM Client to Mbed Cloud Client v2.0.1.1.

@MarceloSalazar @bridadan @dhwalters423 

Test results (had to comment out all `printf` calls in TESTS):

```
+--------------+---------------+-----------------------------------------------+-----------------------------------------+--------+--------+--------+--------------------+
| target       | platform_name | test suite                                    | test case                               | passed | failed | result | elapsed_time (sec) |
+--------------+---------------+-----------------------------------------------+-----------------------------------------+--------+--------+--------+--------------------+
| K64F-GCC_ARM | K64F          | simple-mbed-cloud-client-tests-simple-connect | Connect to network                      | 1      | 0      | OK     | 8.51               |
| K64F-GCC_ARM | K64F          | simple-mbed-cloud-client-tests-simple-connect | Consistent Identity                     | 1      | 0      | OK     | 0.04               |
| K64F-GCC_ARM | K64F          | simple-mbed-cloud-client-tests-simple-connect | Device registration in Device Directory | 1      | 0      | OK     | 1.51               |
| K64F-GCC_ARM | K64F          | simple-mbed-cloud-client-tests-simple-connect | Format storage                          | 1      | 0      | OK     | 24.33              |
| K64F-GCC_ARM | K64F          | simple-mbed-cloud-client-tests-simple-connect | LwM2M GET Test                          | 1      | 0      | OK     | 2.5                |
| K64F-GCC_ARM | K64F          | simple-mbed-cloud-client-tests-simple-connect | LwM2M POST Test                         | 1      | 0      | OK     | 1.07               |
| K64F-GCC_ARM | K64F          | simple-mbed-cloud-client-tests-simple-connect | LwM2M PUT Test                          | 1      | 0      | OK     | 1.44               |
| K64F-GCC_ARM | K64F          | simple-mbed-cloud-client-tests-simple-connect | Pelion Device Management Register       | 1      | 0      | OK     | 5.92               |
| K64F-GCC_ARM | K64F          | simple-mbed-cloud-client-tests-simple-connect | Simple Mbed Cloud Client Initialization | 1      | 0      | OK     | 2.78               |
+--------------+---------------+-----------------------------------------------+-----------------------------------------+--------+--------+--------+--------------------+
mbedgt: test case results: 9 OK
mbedgt: completed in 124.03 sec
```

### CODE REVIEW

File: simple-mbed-cloud-client.cpp [hunk 1]
```diff
@@ -147,22 +163,11 @@ int SimpleMbedCloudClient::init() {
...
-        printf("[SMCC] Firmware storage erasing failed with %" PRId32, status);
-        return 1;
-    }
-#endif
-
-#if MBED_CONF_APP_DEVELOPER_MODE == 1
-    printf("[SMCC] Starting developer flow\n");
-    fcc_status = fcc_developer_flow();
-    if (fcc_status == FCC_STATUS_KCM_FILE_EXIST_ERROR) {
-        printf("[SMCC] Developer credentials already exist\n");
```
Comment (MarceloSalazar) [inline simple-mbed-cloud-client.cpp]: Could you re-enable this message? It's really useful to us to understand whether credentials are reused, or new ones are installed

File: simple-mbed-cloud-client.cpp [hunk 2]
```diff
@@ -147,22 +163,11 @@ int SimpleMbedCloudClient::init() {
     palStatus_t status = PAL_SUCCESS;
     status = pal_fsRmFiles(DEFAULT_FIRMWARE_PATH);
     if(status == PAL_SUCCESS) {
-        printf("[SMCC] Firmware storage erased.\n");
+        tr_info("Firmware storage erased");
```
Comment (MarceloSalazar) [inline simple-mbed-cloud-client.cpp]: Can we get this message enabled by default please?

**Comment by MarceloSalazar** [comment]
> Test results (had to comment out all printf calls in TESTS):

@janjongboom Can you comment what's the problem with the printfs?

**Comment by janjongboom** [comment]
@MarceloSalazar Interfering with greentea, I think Brian suggested to remove all printf's and then they finally worked for me.

File: simple-mbed-cloud-client.cpp (hunk 1 above)
Comment (janjongboom) [reply simple-mbed-cloud-client.cpp]: They're still here, but in `SimpleMbedCloudClient::verify_cloud_configuration` and in a trace call.

File: simple-mbed-cloud-client.cpp (hunk 2 above)
Comment (janjongboom) [reply simple-mbed-cloud-client.cpp]: Done.

**Comment by MarceloSalazar** [comment]
Tests pass after fixing race conditions/timeouts

```
+--------------+---------------+-------------------------------------------------+----------------------------+--------+--------+--------+--------------------+
| target       | platform_name | test suite                                      | test case                  | passed | failed | result | elapsed_time (sec) |
+--------------+---------------+-------------------------------------------------+----------------------------+--------+--------+--------+--------------------+
| K64F-GCC_ARM | K64F          | simple-mbed-cloud-client-tests-dev_mgmt-connect | Connect to Network         | 1      | 0      | OK     | 5.04               |
| K64F-GCC_ARM | K64F          | simple-mbed-cloud-client-tests-dev_mgmt-connect | Consistent Identity        | 1      | 0      | OK     | 2.11               |
| K64F-GCC_ARM | K64F          | simple-mbed-cloud-client-tests-dev_mgmt-connect | Format Storage             | 1      | 0      | OK     | 15.75              |
| K64F-GCC_ARM | K64F          | simple-mbed-cloud-client-tests-dev_mgmt-connect | LwM2M GET Test             | 1      | 0      | OK     | 0.53               |
| K64F-GCC_ARM | K64F          | simple-mbed-cloud-client-tests-dev_mgmt-connect | LwM2M POST Test            | 1      | 0      | OK     | 0.5                |
| K64F-GCC_ARM | K64F          | simple-mbed-cloud-client-tests-dev_mgmt-connect | LwM2M PUT Test             | 1      | 0      | OK     | 0.7                |
| K64F-GCC_ARM | K64F          | simple-mbed-cloud-client-tests-dev_mgmt-connect | LwM2M SET Test             | 1      | 0      | OK     | 0.38               |
| K64F-GCC_ARM | K64F          | simple-mbed-cloud-client-tests-dev_mgmt-connect | Pelion DM Directory        | 1      | 0      | OK     | 6.01               |
| K64F-GCC_ARM | K64F          | simple-mbed-cloud-client-tests-dev_mgmt-connect | Pelion DM Register         | 1      | 0      | OK     | 4.4                |
| K64F-GCC_ARM | K64F          | simple-mbed-cloud-client-tests-dev_mgmt-connect | Simple PDMC Initialization | 1      | 0      | OK     | 2.79               |
+--------------+---------------+-------------------------------------------------+----------------------------+--------+--------+--------+--------------------+
```

We continue to tests with more platforms

**Pull request merged**